content/client-server-api/_index.md

```diff
@@ -336,42 +337,92 @@ to prevent the access token being leaked in access/HTTP logs. The query
 
 ### Relationship between access tokens and devices
 
 Client [devices](../index.html#devices) are closely related to access
-tokens. Matrix servers should record which device each access token is
-assigned to, so that subsequent requests can be handled correctly.
+tokens and refresh tokens. Matrix servers should record which device
+each access token and refresh token are assigned to, so that
+subsequent requests can be handled correctly. When a refresh token is
+used to generate a new access token and refresh token, the new access
+and refresh tokens are now bound to the device associated with the
+initial refresh token.
 
 By default, the [Login](#login) and [Registration](#account-registration-and-management)
 processes auto-generate a new `device_id`. A client is also free to
 generate its own `device_id` or, provided the user remains the same,
 reuse a device: in either case the client should pass the `device_id` in
 the request body. If the client sets the `device_id`, the server will
-invalidate any access token previously assigned to that device. There is
-therefore at most one active access token assigned to each device at any
-one time.
+invalidate any access and refresh tokens previously assigned to that device.
+
+### Refreshing access tokens
+
+{{% added-in v="1.3" %}}
+
+Access tokens can expire after a certain amount of time. Any HTTP
+calls that use an expired access token will return with an error code
+`M_UNKNOWN_TOKEN`, preferably with `soft_logout: true`. When a client
+receives this error and it has a refresh token, it should attempt to
+refresh the access token by calling `/refresh`. Clients can also
+refresh their access token at any time, even if it has not yet
+expired. If the token refresh succeeds, it should use the new token
+for future requests, and can re-try previously-failed requests with
+the new token. When an access token is refreshed, a new refresh token
+may be returned; if a new refresh token is given, the old refresh
+token will be invalidated, and the new refresh token should be used
+when the access token needs to be refreshed.
+
+If the token refresh fails, then the client can treat it as a [soft
+logout](#soft-logout), if the error response included a `soft_logout:
+true` property, and attempt to obtain a new access token by re-logging
+in. If the error response does not include a `soft_logout: true`
+property, the client should consider the user as being logged out.
+
+Handling of clients that do not support refresh tokens is up to the
+homeserver; clients indicate their support for refresh tokens by
+including a `refresh_token: true` property in the request body of the
+`/login` and `/register` endpoints. For example, homeservers may allow
+the use of non-expiring access tokens, or may expire access tokens
+anyways and rely on soft logout behaviour on clients that don't
+support refreshing.
 
 ### Soft logout
 
-When a request fails due to a 401 status code per above, the server can
```

Comment:
this section probably needs `{{% changed-in v="1.3" %}}` ?

Reply:
I've now added the `changed-in` below, in the paragraph that documents the changed behaviour.